### ISSUE TITLE
Running polyA estimation on a custom cDNA protocol

### ISSUE THREAD

**Issue by obegik** [nanoporetech/dorado#624]
We would like to use Dorado polyA estimation for our custom cDNA method called Nano3Pseq, which uses NBD114 kit. The adapter sequence is similar to the VNP oligo used for cDNA synthesis. 

Thanks,
Oguzhan

**Comment by tijyojwad** [comment]
Hi @obegik - thanks for the request! we are working on this feature and hope to have this available as part of the next release (sometime in March)

**Comment by obegik** [comment]
Thanks for the very fast reply! I was going over the files you have on github and under the custom barcodes, there is a test file. Supposedly this could be used when providing custom barcode sequences in case we want to have polyA tail length estimations on single stranded sequencing methods where we only have the adapter/barcode on one side. Do you think I can already test the tail length estimation using custom barcode option and providing the sequences as you did in the custom barcode file? Or should I just wait until the new version is released?

Thanks

**Comment by tijyojwad** [comment]
Hi @obegik - unfortunately that won't work right now since the adapters we look for polyA are hard coded. It should be quite straightforward to edit that and rebuild dorado though. You can put your primer sequences here https://github.com/nanoporetech/dorado/blob/release-v0.5.3/dorado/read_pipeline/PolyACalculator.cpp#L215 and rebuild using the instructions in https://github.com/nanoporetech/dorado/tree/release-v0.5.3?tab=readme-ov-file#clone-and-build and https://github.com/nanoporetech/dorado/blob/release-v0.5.3/DEV.md

**Comment by obegik** [comment]
Thanks @tijyojwad ! I will give it a go and let you know how it goes!

**Comment by obegik** [comment]
I was looking at the script and it seems like you take into account both SSP and VNP, hence the "primer" sequences on both ends of the read. Is that correct? Because in my case I only have VNP, which is only on the forward strand.

**Comment by tijyojwad** [comment]
Yes we take into account both ends to determine the orientation of the cDNA and where to expect the tail.

If it's only single ended for you, can you post a reference diagram for what the construct would look like? is it something like
```
ADAPTER ---- VNP ---- poly T ---- rc( cDNA ) 
```

**Comment by obegik** [comment]
NA----NB---CTTCCGATCACTTGCCTGTCGCTCTATCTTCN---polyT---cDNA

The sequence I shared above is similar to VNP, but not exactly the same (Some nucleotides added on the 5'end). This is the adapter we use for Nano3P-seq.

**Comment by tijyojwad** [comment]
gotcha - okay then I would suggest changing the algorithm the following way - 

In https://github.com/nanoporetech/dorado/blob/release-v0.5.3/dorado/read_pipeline/PolyACalculator.cpp#L214

1. Set VNP to your custom flank sequence
2. comment out line using any of the other sequences (SSP, SSP_rc, VNP_rc)
3. Comment out all edlib calls other than `top_v2` that searches for VNP
4. set `fwd = false`
5. set `proceed` to `top_v2.editDistance < 10` or something low

and that should let you search for your custom construct

**Comment by tijyojwad** [comment]
Hi @obegik we are working on enabling some features to help your use case. would you be open to sharing some sample data so we can test our solution?

**Comment by obegik** [comment]
Hi,

Certainly! Currently I am working on a new set of adapters to improve the prediction and this might take a few weeks to share with you. Otherwise I could share with you the data with old adapters. Whatever works for you!

**Comment by tijyojwad** [comment]
If you're open to it, we'd love to test both :). The adapters and a few reads in a pod5 would be awesome.

**Comment by tijyojwad** [comment]
Yes perfect - I can be reached at joyjit dot daw at nanoporetech dot com . Thanks @obegik !

**Comment by obegik** [comment]
Hi Joyjit, I sent you an email a few days ago. just checking if you received it!

**Comment by malton-ont** [comment]
Closing - dorado has supported custom primers for cDNA polyA estimation since 0.7.0.
